File: CPP/Program/transaksi.cpp
#include <iostream>
#include <string>
#include <vector>
#include "laporanSampah.cpp"

using namespace std;

class Transaksi {
    // Atribut private
    private:
        string idTransaksi;
        string tanggal; 
        vector<LaporanSampah> daftarSampah; // Asosiasi

    public:
        // Constructor kosong
        Transaksi() {
        }

        // Constructor
        Transaksi(string idTransaksi, string tanggal, vector<LaporanSampah> daftarSampah) {
            this->idTransaksi = idTransaksi;
            this->tanggal = tanggal;
            this->daftarSampah = daftarSampah;
        }

        // getter setter id transaksi
        string getIdTransaksi() {return idTransaksi;}
        
        void setIdTransaksi(string idTransaksi) {this->idTransaksi = idTransaksi;}
        
        // getter setter tanggal
        string getTanggal() {return tanggal;}
        
        void setTanggal(string tanggal) {this->tanggal = tanggal;}
        
        // getter setter daftarSampah
        void tambahSampah(LaporanSampah sampah) {
        daftarSampah.push_back(sampah);
        }

        // Getter daftarSampah (return array of object)
        vector<LaporanSampah> getDaftarSampah() { 
            return daftarSampah;
        }

    // Destructor
    ~Transaksi() {    
    }
};

File: PYTHON/Program/transaksi.py
from laporan_sampah import LaporanSampah

class Transaksi:
    def __init__(self, id_transaksi="", tanggal="", daftar_sampah=None):
        self.__id_transaksi = id_transaksi
        self.__tanggal = tanggal
        self.__daftar_sampah = daftar_sampah if daftar_sampah is not None else []

    # getter setter id transaksi
    def get_id_transaksi(self):
        return self.__id_transaksi

    def set_id_transaksi(self, id_transaksi):
        self.__id_transaksi = id_transaksi

    # getter setter tanggal
    def get_tanggal(self):
        return self.__tanggal

    def set_tanggal(self, tanggal):
        self.__tanggal = tanggal

    # menambah laporan sampah (asosiasi)
    def tambah_sampah(self, sampah: LaporanSampah):
        self.__daftar_sampah.append(sampah)

    # Getter daftar sampah (list of object)
    def get_daftar_sampah(self):
        return self.__daftar_sampah
